tests/rptest/tests/cluster_quota_test.py

```diff
@@ -100,3 +102,114 @@ def test_partition_throttle_mechanism(self):
+        """
+        Ensure group rate throttle works
+        """
+
+        self.redpanda.set_cluster_config({
+            "kafka_client_group_byte_rate_quota": [
+                {
+                    "group_name": "group_1",
+                    "clients_prefix": "producer_group_alone_producer",
+                    "quota": 10240
+                },
+                {
+                    "group_name": "group_2",
+                    "clients_prefix": "producer_group_multiple",
+                    "quota": 10240
+                },
+            ],
+            "max_kafka_throttle_delay_ms":
+            10,
+        })
+
+        producer = KafkaProducer(
+            acks="all",
+            bootstrap_servers=self.redpanda.brokers_list(),
+            value_serializer=str.encode,
+            retries=1,
+            client_id="producer_group_alone_producer")
+
+        self.msg = "".join(
+            random.choice(string.ascii_lowercase) for _ in range(1024))
+
+        # Produce under the limit
+        self.produce(producer, 10)
+        assert producer.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] == 0
+
+        # Produce more than limit
+        self.produce(producer, 100)
+        assert producer.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] > 0
+
+        producer_1 = KafkaProducer(
+            acks="all",
+            bootstrap_servers=self.redpanda.brokers_list(1),
+            value_serializer=str.encode,
+            retries=1,
+            client_id="producer_group_multiple_1")
+
+        producer_2 = KafkaProducer(
+            acks="all",
+            bootstrap_servers=self.redpanda.brokers_list(1),
+            value_serializer=str.encode,
+            client_id="producer_group_multiple_2")
+
+        # Produce under the limit
+        self.produce(producer_1, 10)
+        assert producer_1.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] == 0
+
+        # Produce more than the limit
+        self.produce(producer_1, 100)
+
+        # Produce under the limit for client, but more than limit for group
+        self.produce(producer_2, 10)
+        assert producer_1.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] > 0
+        assert producer_2.metrics(
+        )["producer-metrics"]["produce-throttle-time-max"] > 0
+
+        self.redpanda.set_cluster_config({
+            "kafka_client_group_byte_rate_quota": {
+                "group_name": "change_config_group",
+                "clients_prefix": "new_producer_group",
+                "quota": 1024
+            }
+        })
+
+        self.logger.error(self.redpanda.brokers_list(1)[0])
+        producer = KafkaProducer(
+            acks="all",
+            bootstrap_servers=self.redpanda.brokers_list(1),
+            value_serializer=str.encode,
+            client_id="new_producer_group_producer")
+
+        self.produce(producer, 100)
+
+        assert producer.metrics(
```

Comment:
Does the producer throttle by the `throttle_time_ms` response returned from the broker?

Reply:
It is throttled inside redpanda.
Message is delayed
What producer will do with it - depends on realization.

Reply:
Currently all throttling done fills the `throttle_time_ms` field, that is what it is for. Redpanda throttling the next request is the second best option that we do in case a badly behaving client does not respect the `throttle_time_ms` value.

FWIW I think this field is getting automatically filled by the quota manager and/or response handler, can we just verify that?